Project/src/main/java/app/Application.java

```diff
@@ -24,33 +23,35 @@ public static void main(String[] args) {
 
     //Temporary test data
     @Bean
-    public CommandLineRunner demo(FileRepository fileRepository, UserRepository userRepository, InMemoryUserDetailsManager inMemoryUserDetailsManage) {
+    public CommandLineRunner demo(ArticleRepository articleRepository, RoleRepository roleRepository, UserRepository userRepository, InMemoryUserDetailsManager inMemoryUserDetailsManage) {
         return (args) -> {
             PasswordEncoder encoder = PasswordEncoderFactories.createDelegatingPasswordEncoder();
             // save a couple of customers
-            File file = new File();
-            User user1 = new User("test1", encoder.encode("password"), Role.SUBMITTER.toString());
-            User user2 = new User("test2", encoder.encode("password"), Role.EDITOR.toString());
-            User user3 = new User("test3", encoder.encode("password"), Role.REVIEWER.toString());
+            Article article = new Article();
+            User user1 = new User("test1", encoder.encode("password"), Role.submitter);
+            User user2 = new User("test2", encoder.encode("password"), Role.editor);
+            User user3 = new User("test3", encoder.encode("password"), Role.reviewer);
 
-            file.setFileName("FileTest1");
+            article.setFileName("FileTest1");
             List<User> users = new ArrayList<User>();
             users.add(user1);
             users.add(user3);
-            file.setUser(users);
-            file.setStatus(ArticleStatus.SUBMITTED);
-            user1.addFiles(file);
+            article.setAuthorizedUsers(users);
+            user1.addAuthorizedArticle(article);
```

Comment:
Redundant, the article should know who it's submitter and reviewer are, and they can find all their articles by searching by user.

Reply:
the submitter is only added to authorized users in the controller, so elsewhere (like in this test code) the submitter also needs to be added explicitly. Likewise, reviewer can only be added by an editor (which it wasn’t intended to be here).

I can add a submitter User to the constructor to remove the addAuthorizedUser calls.

Reply:
What I'm saying is you should have an article.setSubmitter and article.setReviewer only, you shouldn't need a user.addAuthorizedArticle, as you can get the other two links from the article itself